tests/integration/workflows/test_workflow_actions.py

```diff
@@ -329,3 +400,53 @@ def test_normalize_journal_titles_unknown_journals_no_ref(workflow_app, insert_j
     assert obj.data['publication_info'][2]['journal_title'] == 'Unknown2'
     assert 'journal_record' not in obj.data['publication_info'][0]
     assert 'journal_record' not in obj.data['publication_info'][2]
+
+
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.arxiv.download_file_to_workflow',
+    side_effect=fake_download_file,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.actions.download_file_to_workflow',
+    side_effect=fake_download_file,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.arxiv.is_pdf_link',
+    return_value=True
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.beard.json_api_request',
+    side_effect=fake_beard_api_request,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.magpie.json_api_request',
+    side_effect=fake_magpie_api_request,
+)
+def test_refextract_from_pdf(
+    mocked_api_request_magpie,
+    mocked_api_request_beard,
+    mocked_is_pdf_link,
+    mocked_package_download,
+    mocked_arxiv_download,
+    insert_cited_record,
+    workflow_app,
+    mocked_external_services
+):
+    """Test refextract by going through the entire workflow."""
+
+    citing_record, categories = insert_citing_record()
+
+    extra_config = {
+        "BEARD_API_URL": "http://example.com/beard",
+        "MAGPIE_API_URL": "http://example.com/magpie",
+        'ARXIV_CATEGORIES': categories,
+    }
+
+    with workflow_app.app_context():
+        with mock.patch.dict(workflow_app.config, extra_config):
+            citing_doc_workflow_uuid = start('article', [citing_record])
+
+        citing_doc_eng = WorkflowEngine.from_uuid(citing_doc_workflow_uuid)
+        citing_doc_obj = citing_doc_eng.processed_objects[0]
+
+        assert citing_doc_obj.data['references'][7]['record'] == get_record_ref(insert_cited_record['control_number'], 'literature')
```

Comment:
I think that here we should follow the suggestion in https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html:
> Tests can avoid complexity by stating their inputs and outputs directly rather than computing them. Otherwise it's easy for tests to develop their own bugs.